src/Model/Entity/BaseEntity.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Friendica\Model\Entity;
+
+use ArrayAccess;
+use Friendica\Network\HTTPException;
+
+abstract class BaseEntity implements ArrayAccess
+{
+	/**
+	 * Model record abstraction.
+	 * Child classes never have to interact directly with it.
+	 * Please use the magic getter instead.
+	 *
+	 * @var array
+	 */
+	protected $data = [];
+
+	/**
+	 * Private constructor of the entity
+	 * Entities are not meant to get created manually by hand
+	 *
+	 * Instead use the static create() function
+	 *
+	 * @param array $data
+	 */
+	private function __construct(array $data = [])
```

Comment:
making the constructor `private` means that it is impossible to create entities without using:
- `BaseEntity::create()`
- `BaseEntity::createFromPrototype()`